### PR TITLE
Session windows and serialization conversion

### PR DESCRIPTION
**Pull request by davetroiano**

### CODE REVIEW

File: session-windows/kstreams/src/main/java/io/confluent/developer/SessionWindow.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        Properties properties;
+        if (args.length > 0) {
+            properties = Utils.loadProperties(args[0]);
+        } else {
+            properties = Utils.loadProperties();
+        }
+        properties.put(StreamsConfig.APPLICATION_ID_CONFIG, "session-window-application");
+
+        ClicksDataGenerator generator = new ClicksDataGenerator(properties);
+        generator.generate();
```
Comment (bbejeck) [inline session-windows/kstreams/src/main/java/io/confluent/developer/SessionWindow.java:81]: You can delete these two lines.

File: session-windows/kstreams/src/main/java/io/confluent/developer/SessionWindow.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            // For local running only; don't do this in production as it wipes out all local state
+            kafkaStreams.cleanUp();
+            kafkaStreams.start();
+            countDownLatch.await();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    static class ClicksDataGenerator {
```
Comment (bbejeck) [inline session-windows/kstreams/src/main/java/io/confluent/developer/SessionWindow.java:102]: You can delete the `ClicksDataGenerator` class

File: session-windows/kstreams/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                String sessionInfo = String.format("Session info started: %s ended: %s with count %s", start, end, count);
+                return KeyValue.pair(windowedKey.key(), sessionInfo);
+            })
+            .to(OUTPUT_TOPIC, Produced.with(Serdes.String(), Serdes.String()));
+```
+
+Let's review the key points in this example.
+
+``` java
+    .groupByKey(Grouped.with(Serdes.String(), movieRatingSerde))
```
[the remark this answers is not in the document]
Comment (davetroiano) [reply session-windows/kstreams/README.md]: made this change in follow on PR

**Review by bbejeck — APPROVED** [review]
LGTM!

**Pull request merged**